### ISSUE TITLE
Remove unneeded generics in GraphQLRequest and GraphQLOperation

### ISSUE THREAD

**Issue by jamesblasco** [aws-amplify/amplify-flutter#1270]
**Describe the bug**
I don't think there is a use for the GraphQLRequest generic, and I think it should removed from the api as it only adds unneeded complexity.

Instead of
```dart
 GraphQLOperation<T> query<T>({required GraphQLRequest<T> request}) {
````

do 
```dart
 GraphQLOperation query({required GraphQLRequest request}) {
````
<img width="612" alt="Screenshot 2022-01-14 at 14 01 18" src="https://user-images.githubusercontent.com/19904063/149519385-15b5b358-19f5-484a-a80c-149abf01c363.png">

**Comment by dnys1** [comment]
Hi @jamesblasco - again, great callout. The generics do not serve a purpose at the moment - in the future, `T` will be able to be used for Model types (from DataStore), to do API category calls which return deserialized models. Hope this helps!

**Comment by jamesblasco** [comment]
Ah makes see! Thank you for the explanation :)

**Comment by dnys1** [comment]
Absolutely :D 

I will leave this issue open because I think we should document it better.

**Comment by ragingsquirrel3** [comment]
We recently released model-based GraphQL helpers with 0.4.0 and I think these generics might make a bit more sense now. Like you see in examples on https://docs.amplify.aws/lib/graphqlapi/query-data/q/platform/flutter/ the type passed to generic can now be a model instance, like `Todo` instead of always being a string.

```dart
final request = ModelQueries.get(
      Todo.classType, createdTodo.id);
```

in this case, the `request` will be `GraphQLRequest<Todo>`. Further, the response from the plugin will have the same type as the request `GraphQLResponse<Todo>` where `response.data` would be of type `Todo`. I hope this helps explain why there are generics here. There are some new dart doc comments on the request class that try to document this but let us know if this can be made more clear.

Going to close this issue with that explanation.
